File: project_python/playgame_เป่ายิงฉุบ.py
### เป่ายิงฉุบ
import random


choice = ["hammer", "scissor", "paper"]

def play_game():
    print("Welcome to เป่ายิงฉุบ")
    print("Type 'stop' to quit game.\n")
    user_name = input("What is your name? : ")
    print(f"Are you ready? You have 3 choices: {choice}")

    user_point = 0
    computer_point = 0


    while True:
        user_select = input("\nYou choose one: ")
        computer_select = random.choice(choice)

        if user_select == "stop":
            if user_point > computer_point:
                print("\nCongratulation! You win!! 🥳 \nScore".upper())
            elif user_point < computer_point:
                print("\nSorry, You lost!! 😭\nSCORE".upper())
            else:
                print("\nWow, Tie!! 🙌🏻\nScore".upper())


            return f"{user_name} {user_point} : {computer_point} Computer"

        else:
            if user_select == computer_select:
                print(f"Computer chose: {computer_select}")
                print("tie!")
            elif user_select == "hammer" and computer_select == "scissor":
                print(f"Computer chose: {computer_select}")
                print("You win!")
                user_point += 1
            elif user_select == "hammer" and computer_select == "paper":
                print(f"Computer chose: {computer_select}")
                print("You lost!")
                computer_point += 1
            elif user_select == "scissor" and computer_select == "paper":
                print(f"Computer chose: {computer_select}")
                print("You win!")
                user_point += 1
            elif user_select == "scissor" and computer_select == "hammer":
                print(f"Computer chose: {computer_select}")
                print("You lost!")
                computer_point += 1
            elif user_select == "paper" and computer_select == "scissor":
                print(f"Computer chose: {computer_select}")
                print("You lost!")
                computer_point += 1
            elif user_select == "paper" and computer_select == "hammer":
                print(f"Computer chose: {computer_select}")
                print("You win!")
                user_point += 1
            else:
                print(f"\n❌ Sorry, Please check your spelling. \nChoose again!: {choice}")

play_game()

File: project_R/playgame_เป่ายิงฉุบ.R
### เป่ายิงฉุบ

choice <- c("hammer", "scissor", "paper")

play_game <- function(){
  print("Welcome to เป่ายิงฉุบ")
  print("Type 'stop' to quit game.")
  user_name <- readline("What is your name? : ")
  print(paste("Are you ready? You have 3 choices:"))
  print(choice)
  
  user_point <- 0
  computer_point <- 0
  
  
  while(TRUE) {
    user_select <- readline("Choose one: ")
    computer_select <- sample(choice,1)
    
    if (user_select == "stop"){

      if(user_point > computer_point){
        print("Congratulation! You win!!")
      } else if (user_point < computer_point){
        print("Sorry, You lost!!")
      } else {
        print("Score Tie!!")
      }
      
      print(paste(user_name, user_point, ":", computer_point, "Computer"))
      break
    } else {
      
      
      if (user_select == computer_select){
        print(computer_select)
        print("tie!")
    } else if (user_select == "hammer" & computer_select == "scissor"){
      print(computer_select)
      print("You win!")
      user_point <- user_point + 1
    } else if (user_select == "hammer" & computer_select == "paper"){
      print(computer_select)
      print("You lost!")
      computer_point <- computer_point + 1
    } else if (user_select == "scissor" & computer_select == "paper"){
      print(computer_select)
      print("You win!")
      user_point <- user_point + 1
    } else if (user_select == "scissor" & computer_select == "hammer"){
      print(computer_select)
      print("You lost!")
      computer_point <- computer_point + 1
    } else if (user_select == "paper" & computer_select == "scissor"){
      print(computer_select)
      print("You lost!")
      computer_point <- computer_point + 1
    } else if (user_select == "paper" & computer_select == "hammer"){
      print(computer_select)
      print("You win!")
      user_point <- user_point + 1
    } else {
      print("Sorry, Please try again!")
      print(choice)
    }
      }
  }
}

play_game()
